src/app/exercise/exercises/TriadInversionExercise/triadInversionExercise.ts

```diff
@@ -58,11 +59,26 @@ export const triadInversionExercise = () => {
       const invertionOptions: TriadInversion[] = [0, 1, 2].filter(invertionOption => settings.includedAnswers.includes(triadInversions[invertionOption]));
       const randomTriadInversion: TriadInversion = randomFromList(invertionOptions);
       const answer = triadInversions[randomTriadInversion];
-      const voicing: Note[] = new Chord(randomChordInC).getVoicing({
+      let voicing: Note[] = new Chord(randomChordInC).getVoicing({
         topVoicesInversion: randomTriadInversion,
         withBass: false,
         octave: 3, // picking a lower octave as a high one is more difficult
       });
+      let temp: Note[];
+      const root: Note = Array.from(voicing)[(3 - randomTriadInversion) % 3];
```

Comment:
I worry about after reverse array, the root note is calculated incorrectly. 
what do you think about?

Reply:
Ahhh got you. I did not think about it. Then yeah, keep it. But you can remove the Aray.from since you are just reading from the array, not mutating it, so copying is not needed.

Reply:
@ShacharHarshuv I find if not cloning, array reverse will impact the root note